### PR TITLE
Add signatures for GD image*() returning strings.

### PR DESCRIPTION
**Pull request by javajawa**
When the image* output functions, such as [`imagepng()`](https://www.php.net/manual/en/function.imagepng.php), are passed `null` as the filename/to parameter, the raw image data is returned from the function. 

This patch adds this signature for:
 - `imagebmp`
 - `imagegd`
 - `imagegd2`
 - `imagegif`
 - `imagejpeg`
 - `imagepng`
 - `imagewbmp`
 - `imagewebp`
 - `imagexbm`

### CODE REVIEW

**Comment by javajawa** [comment]
I do not know what to make of the tests failing specifically in php7.4, nor do I see the connection between the binaryOps test and the changes I've made. Suggestions most welcome.

File: src/Reflection/SignatureMap/functionMap.php [hunk 1]
```diff
@@ -4593,15 +4594,19 @@
 'imageftbbox' => ['array', 'size'=>'float', 'angle'=>'float', 'font_file'=>'string', 'text'=>'string', 'extrainfo='=>'array'],
 'imagefttext' => ['array', 'im'=>'resource', 'size'=>'float', 'angle'=>'float', 'x'=>'int', 'y'=>'int', 'col'=>'int', 'font_file'=>'string', 'text'=>'string', 'extrainfo='=>'array'],
 'imagegammacorrect' => ['bool', 'im'=>'resource', 'inputgamma'=>'float', 'outputgamma'=>'float'],
-'imagegd' => ['bool', 'im'=>'resource', 'filename='=>'string|resource|null'],
-'imagegd2' => ['bool', 'im'=>'resource', 'filename='=>'string|resource|null', 'chunk_size='=>'int', 'type='=>'int'],
+'imagegd' => ['bool', 'im'=>'resource', 'filename='=>'string|resource'],
+'imagegd\'1' => ['string', 'im'=>'resource', 'filename='=>'null'],
```
Comment (ondrejmirtes) [inline src/Reflection/SignatureMap/functionMap.php:4598]: Can you test that ParametersAcceptorSelector will figure this out? Add a new data provider to NodeScopeResolverTest::testFileAsserts and see that the right imagegd() calls lead to bool or string with assertType(), thanks.

File: tests/PHPStan/Analyser/data/graphics-draw-return-types.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+use function PHPStan\Analyser\assertType;
+
+$image = imagecreatetruecolor(1, 1);
+$memoryHandle = fopen('php://memory', 'w');
+
+assertType('bool', imagegd($image, 'php://memory'));
+assertType('bool', imagegd($image, $memoryHandle));
+assertType('string|false', imagegd($image, null));
```
Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/data/graphics-draw-return-types.php:10]: Please verify `imagegd($image);` too, thanks.

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by javajawa** [comment]
> Thank you!

Thank you for the help, hints, and frankly the entire project!

**Pull request merged**